### ISSUE TITLE
SA1000: Spacing around `new()` expression inconsistent with formatter support

### ISSUE THREAD

**Issue by tillig** [DotNetAnalyzers/StyleCopAnalyzers#3311]
In C# 9.0, you can omit the type name when using `new` in some cases:

```c#
// With the type name
List<int> ws = new List<int>();

// Without the type name
List<int> xs = new();
List<int> ys = new(capacity: 10_000);
List<int> zs = new() { Capacity = 20_000 };
```

The SA1000 rule catches `new()` and wants a space like `new ()` because it assumes the next thing should be a type name. It should treat `new()` like a function instead of like `new TypeHere()` with respect to spacing. That would also allow C# formatters, which treat `new()` like a function, to be consistent with StyleCop rules. (Right now they fight a bit.)

**Comment by tillig** [comment]
It looks like [there's already a unit test around this](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/32818bec4ea7c8ee18c272259cac585506e4ce43/StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp9/SpacingRules/SA1000CSharp9UnitTests.cs#L17) so perhaps this just hasn't been released yet?

**Comment by tillig** [comment]
I'm dumb and need to RTFM. Updated to 1.2.0-beta.321 and all is well. Sorry for any confusion.

**Comment by sharwell** [comment]
Duplicate of #3214
